app/src/main/java/com/github/se/polyfit/ui/components/selector/DateSelector.kt

```diff
@@ -0,0 +1,110 @@
+package com.github.se.polyfit.ui.components.selector
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.github.se.polyfit.ui.theme.PrimaryPink
+import com.github.se.polyfit.ui.theme.PurpleGrey40
+import java.time.LocalDate
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DateSelector(onConfirm: (LocalDate) -> Unit, modifier: Modifier = Modifier) {
+  var showDatePicker by remember { mutableStateOf(false) }
+
+  // I'm *pretty sure* this should select the right time and zone, but if test flakes this is why
+  val initialMillis = LocalDate.now().atStartOfDay().toInstant(ZoneOffset.UTC).toEpochMilli()
```

Comment:
For example, if the user is in a time zone that is a day behind UTC, this would cause the "initial start date" (`initialMillis`) to be incorrect. However, this issue is relatively minor.

Reply:
Changed to use device time, which, for some reason is giving me the wrong date, but i think it may be due to my computer having north America setting